test/unit/matrix/test_util_matrix.cpp

```diff
@@ -138,39 +138,38 @@ void check_is_hermitian(Matrix<const T, Device::CPU>& matrix, comm::Communicator
         continue;
 
       if (current_rank == owner_original) {
-        const auto& tile_original = sync_wait(matrix.read_sender(index_tile_original)).get();
-        pika::shared_future<Tile<const T, Device::CPU>> tile_transposed;
+        const auto& tile_original = sync_wait(matrix.read(index_tile_original)).get();
```

Comment:
Yep, good catch! I think it happens to work right now because we're only doing read-only access and using the tile right away, but it could break at any time.